### DISCUSSION TITLE
Fulfillment completion process

### DISCUSSION THREAD

**Discussion by shunnmugam** [medusajs/medusa#4528] [Q&A]
is there any method for Fulfillment complete, if no, how to mark product as delivered?

**Comment by srindom** [comment]
Could you provide a bit more detail to what you are looking for?

You can create a fulfillment on an order with the order's line items. Creating a fulfillment is signifying that the order has been placed, accepted and is now pending shipment. When the fulfillment is shipped it can be marked as shipped.

Is this what you are looking for?

**Reply by shunnmugam** [reply]
Hello @srindom, thanks for your response, have some more queries.

- When we need to mark the product status as completed?

- is it completed once the product is shipped or delivered to customer?

- how to identify the product is delivered to customer?

**Comment by thejspro** [comment]
After creating a fulfilment, order status still says "Processing".

I would also like to know how to mark order as "Completed" in the admin panel.

**Reply by HalimACeylan** [reply]
There is a ticket in the discord page(here is the solution I think)<img width="717" alt="image" src="https://github.com/medusajs/medusa/assets/64225727/19f3cf31-e998-45b3-a848-afc0fa9a43ef">

**Reply by aldo-roman** [reply]
For anyone running into this, I solved it by creating a subscriber to complete the order:

```ts
// src/subscribers/complete-order.ts

import { OrderService, PaymentStatus } from "@medusajs/medusa";

class CompleteOrderSubscriber {

  orderService: OrderService;

  constructor({ eventBusService, orderService }) {
    this.orderService = orderService;
    eventBusService.subscribe(
      OrderService.Events.SHIPMENT_CREATED,
      this.onShipmentCreated
    );
    // Added payment_captured event, as shipment can happen before payment by default
    eventBusService.subscribe(
      OrderService.Events.PAYMENT_CAPTURED,
      this.onShipmentCreated
    );
  }

  onShipmentCreated = async ({ id }) => {
    const order = await this.orderService.retrieve(id, {
      relations: ["items"],
    });

    const isPaymentCaptured = order.payment_status === PaymentStatus.CAPTURED;

    const areAllItemsShipped = order.items.every(
      (i) => i.shipped_quantity === i.quantity
    );

    if (isPaymentCaptured && areAllItemsShipped) {
      this.orderService.completeOrder(id);
    }
  };
}

export default CompleteOrderSubscriber;
```

**Comment by savire** [comment]
I don't see any option for the buyers(users) to complete an order though. Weird really.

**Reply by namnguyenn17** [reply]
Did you solve for this problem yet? I have the same like you as well

**Reply by savire** [reply]
Hi mate,
I was just testing it out and have decided to move on with other simpler open source product since got no replies here.

**Reply by freakk101** [reply]
> Hi mate, I was just testing it out and have decided to move on with other simpler open source product since got no replies here.

which opensource project you moved to ?
